src/test/groovy/com/example/demo/LocationTests.groovy

```diff
@@ -0,0 +1,192 @@
+        def loc = "Cork"
+        def e1 = locationRepository.save(new LocationEntity(loc))
+        when: "a valid call to addLocation is made"
+        def except = mockMvc.perform(get("/api/location/findById?id=2")).andReturn().getResolvedException().getClass()
+        then: "response status exception thrown"
+        assert(except.is(ResponseStatusException))
+    }
+
+    def "LocationController - REST - valid deleteById"() {
+        given: "an entity"
+        def loc = "Cork"
+        def e1 = locationRepository.save(new LocationEntity(loc))
+        when: "a valid call to deleteById is made"
+        def result = mockMvc.perform(delete("/api/location/deleteById?id=1")).andReturn().response.contentAsString
+        then: "result verifies the delete"
+        result == "location deleted successfully."
+    }
+
+    def "LocationController - REST - invalid deleteById"() {
+        given: "an entity"
+        def loc = "Cork"
+        def e1 = locationRepository.save(new LocationEntity(loc))
+        when: "an invalid call to deleteById is made"
+        def except = mockMvc.perform(delete("/api/location/deleteById?id=2")).andReturn().getResolvedException().getClass()
+        then: "response status exception thrown"
+        assert(except.is(ResponseStatusException))
+    }
+
+    def "LocationController - REST - Page"() {
+        given: "an entity"
+        def loc1 = "Belfast"
+        def loc2 = "Cork"
+        def loc3 = "Portrush"
+        def e1 = locationRepository.save(new LocationEntity(loc1))
+        def e2 = locationRepository.save(new LocationEntity(loc2))
+        def e3 = locationRepository.save(new LocationEntity(loc3))
+        when: "a valid call to listPageable is made"
+        def result = mockMvc.perform(get("/api/location/listPageable?page=0&size=3&sort=location,desc")).andReturn().response.contentAsString
+        def json = new JsonSlurper().parseText(result)
+        then: "result verifies the returned pageable"
+        json.totalPages == 1
+        json.totalElements == 3
+        json.content[0].location == "Portrush"
+        json.content[1].location == "Cork"
+        json.content[2].location == "Belfast"
+    }
+
+    def "LocationController - REST - Page2"() {
+        given: "an entity"
+        def loc1 = "Belfast"
+        def loc2 = "Cork"
+        def loc3 = "Portrush"
+        def e1 = locationRepository.save(new LocationEntity(loc1))
+        def e2 = locationRepository.save(new LocationEntity(loc2))
+        def e3 = locationRepository.save(new LocationEntity(loc3))
+        when: "a valid call to listPageable is made"
+        def result = mockMvc.perform(get("/api/location/listPageable?page=0&size=1&sort=location,asc")).andReturn().response.contentAsString
+        def json = new JsonSlurper().parseText(result)
+        then: "result verifies the returned pageable"
+        json.totalPages == 3
+        json.totalElements == 3
+        json.content[0].location == "Belfast"
```

Comment:
Should probably also assert `json.content.size() == 1` to prove you got a page of size 1